query_select.go

```diff
@@ -538,6 +538,11 @@ func (q *SelectQuery) appendQuery(
 	if count && !cteCount {
 		b = append(b, "count(*)"...)
 	} else {
+		// Allows Limit() without Order() with MSSQL as per https://stackoverflow.com/a/36156953
+		if q.limit > 0 && fmter.Dialect().Features().Has(feature.OffsetFetch) && len(q.order) == 0 {
+			b = append(b, "0 AS _temp_sort, "...)
+			q.order = []schema.QueryWithArgs{schema.UnsafeIdent("_temp_sort")}
```

Comment:
This is query generation and it should not change the original `SelectQuery`, because someone may be reusing them.

You probably need to add another block of code to generate order without changing the `SelectQuery`.

Reply:
Do you mean that if I have a `SelectQuery` in a variable `a` and then do `b := a.Limit(5)`, the value of `a` would change, and that would be a problem? Isn't that also the case without the proposed change? I cannot seem to find an analog for this case to compare to. Do you mean something like `ORDER BY` after `LIMIT`, or is there another case where there could be something appended to a select query after `Limit()`? An example would be much appreciated.

Reply:
@CL-Jeremy I've fixed this myself. See the commits in this branch.